src/MongoDB.Driver.Legacy/MongoServerSettings.cs

```diff
@@ -727,7 +788,17 @@ public static MongoServerSettings FromClientSettings(MongoClientSettings clientS
             serverSettings.ApplicationName = clientSettings.ApplicationName;
             serverSettings.ClusterConfigurator = clientSettings.ClusterConfigurator;
             serverSettings.Compressors = clientSettings.Compressors;
-            serverSettings.ConnectionMode = clientSettings.ConnectionMode;
+#pragma warning disable CS0618
+            serverSettings._connectionModeSwitch = clientSettings.ConnectionModeSwitch;
```

Comment:
Remove this line. The connection mode switch will be set indirectly via lines 795 or 799.